specs/bond-manager.md

```diff
@@ -0,0 +1,146 @@
+But in more complex games, such as the fault dispute game, challengers and defenders
+perform a series of alternating onchain transactions requiring bonds at each step.
+
+## The Bond Problem
+
+At its core, the bond manager is straightforward - it escrows or holds ether and can be claimed
+at maturity or seized if forfeited. But the uncertainty of introducing bonds lies in the
+bond _sizing_, i.e. how much should a bond be? Sizing bonds correctly is a function of
+the bond invariant: the bond must be greater than or equal to the cost of the next step.
+If bonds are priced too low, then the bond invariant is violated and there isn't an economic
+incentive to execute the next step. If bonds are priced too high, then the actors posting
+bonds can be priced out.
+
+Below, we outline two different approaches to sizing bonds and the tradeoffs of each.
+
+### Simple Bond
+
+The _Simple Bond_ is a very conservative approach to bond management, establishing a **fixed** bond
+size. The idea behind simple bond pricing is to establish the worst case gas cost for
+the next step in the dispute game.
+
+With this approach, the size of the bond is computed up-front when a dispute game is created.
+For example, in an attestation dispute game, this bond size can be computed as such:
+
+```md
+bond_size = (signer_threshold * (challenge_gas + security overhead)) + resolution_gas(signer_threshold)
+```
+
+Notice that since the bond size is linearly proportional to the number of signers, the economic
+security a given bond size provides decreases as the number of signers increases. Also note, the
+`resolution_gas` function is split out from the `challenge_gas` cost because only the _last_ challenger
+will pay for the gas cost to resolve the game in the attestation dispute game.
+
+Working backwards, if we assume the number of signers to be `5`, a negligible resolution gas cost, and
+a `100,000` gas cost to progress the game, then the bond size should cover `500,000` gas. Meaning, a bond
+of `1 ether` would cover the cost of progressing the game for `5` signers as long as the gas price
+(base fee) does not exceed `2,000 gwei` for the entire finalization window. It would be prohibitively
+expensive to keep the settlement layer base fee this high.
+
+### Variable Bond
+
+Better bond heuristics can be used to establish a bond price that accounts for
+the time-weighted gas price. One instance of this called _Varable Bonds_ use a
+separate oracle contract, `GasPriceFluctuationTracker`, that tracks gas fluctuations
+within a pre-determined bounds. This replaces the ideal solution of tracking
+challenge costs over all L1 blocks, but provides a reasonable bounds. The initial
+actors posting this bond are responsible for funding this contract.
+
+## Contract Interface
+
+Below is a minimal interface for the bond manager contract.
+
+```solidity
+/**
+ * @title IBondManager
+ * @notice The IBondManager is an interface for a contract that handles bond management.
```

Comment:
I don't think we need the word bond in any of the functions because its the bond manager. This is a personal preference but `seize` feels cleaner to me than `seizeBond`. Same for each of the other function names

Reply:
100% good catch